rlclientlib/logger/event_logger.h

```diff
@@ -170,30 +170,40 @@ class slates_logger : public event_logger<slates_decision_event> {
     int report_action_taken(const char* event_id, api_status* status);
   };
 
-  class generic_event_logger : public event_logger<generic_event> {
+  enum compression_mode_enum
```

Comment:
This should be an enum class so that is namespaced and type checked properly
```suggestion
  enum class compression_mode_enum
```